lib/bundler/cli.rb

```diff
@@ -34,8 +34,14 @@ def initialize(*args)
       super
 
       custom_gemfile = options[:gemfile] || Bundler.settings[:gemfile]
+
+      if ENV["BUNDLE_GEMFILE"] && !ENV["BUNDLE_GEMFILE"].empty?
```

Comment:
We don't need to set `custom_gemfile` with `ENV["BUNDLE_GEMFILE"]`  if we're going to skip using it using `unless ENV["BUNDLE_GEMFILE"]`

Reply:
Then we will have to move `Bundle.reset_paths!` outside if condition to run it in any case?